### PR TITLE
documents a peculiarity on `on`

### PR DESCRIPTION
**Pull request by titoBouzout**

### CODE REVIEW

File: langs/en/api.md [hunk 1]
```diff
@@ -536,6 +536,22 @@ createEffect(on(a, (v) => console.log(v), { defer: true }));
...
+Please note that on nested objects, adding or removing a property from the parent object will trigger an effect. This is done to support Iteraction on objects, like Array.
+
+```js
+let state = createMutable({ nested: { a: 1 } });
+
+createEffect(
+  on(
+    () => state.nested.a,
+    () => console.log("could be unexpected"),
+    { defer: true }
```
Comment (thetarnav) [inline langs/en/api.md]: Defer isn't necessary here, right? Just don't want for it to confuse somebody that this happens only with the use of `defer`. 
```suggestion
    () => console.log("could be unexpected")
```

Comment (titoBouzout) [reply langs/en/api.md]: Yes, Thanks!

**Review by edemaine — COMMENTED** [review]
I wonder if this belongs in the sections on Stores and Mutables, not `on`. ("Nested objects" at least should instead refer to these things.) This is a general dependency issue, not just `on`, though I agree it's particularly weird with `on`.

**Comment by titoBouzout** [comment]
I agree, thought it may be more suitable for an "Addendum" section on "Reactivity", explaining this peculiarity in detail and with more examples, ex this update focus on the unexpected not on when that behaviour is useful.  The first paragraph introduces reactivity, explaining this there could be just too complicated/turn off. 
In case something like that gets added, it would be nice to at least keep a Note on the `òn` section because this is "surprising".

**Comment by edemaine** [comment]
My thinking was, if someone isn't using stores or immutables, they don't need to know this. So it's kind of confusing to include in `on` (though I could see adding a pointer from there). I'd rather have it as a "gotcha" under stores/immutables.

**Comment by titoBouzout** [comment]
Moved it to `createMutable`  and attempted to include a link from `on`

**Pull request merged**